bokeh/cli/core.py

```diff
@@ -118,31 +145,82 @@ def __init__(self, input_source, output, title, chart_type, series, palette,
+            title, output, show_legend=self.show_legend
+        )
+        if self.smart_filters:
+            self.chart_args['tools'] = "pan,wheel_zoom,box_zoom,reset,save," \
+                                       "box_select,lasso_select"
+
+        if map_:
+            self.map_options['lat'], self.map_options['lng'] = \
+                [float(x) for x in map_.strip().split(',')]
+
+            self.map_options['zoom'] = int(map_zoom)
+            # Yeah, unfortunate namings.. :-)
+            self.map_options['map_type'] = map_layer
+
+    def on_selection_changed(self, obj, attrname, old, new):
+        self.current_selection = new
+
+    def limit_source(self, source):
+        """ Limit source to cli.window_size, if set.
+
+        Args:
+            source (mapping): dict-like object
+        """
+        if self.window_size:
+            for key in source.keys():
+                source[key] = source[key][-self.window_size:]
 
     def run(self):
         """ Start the CLI logic creating the input source, data conversions,
         chart instances to show and all other niceties provided by CLI
         """
         try:
+            self.limit_source(self.source)
+
+            children = []
+            if self.smart_filters:
+                copy_selection = Button(label="copy current selection")
+                copy_selection.on_click(self.on_copy)
+                children.append(copy_selection)
+
             self.chart = create_chart(
-                self.series, self.source, self.index, self.factories, **self.chart_args
+                self.series, self.source, self.index, self.factories,
+                self.map_options, children=children, **self.chart_args
             )
             self.chart.show()
 
             self.has_ranged_x_axis = 'ranged_x_axis' in self.source.columns
             self.columns = [c for c in self.source.columns if c != 'ranged_x_axis']
 
+            if self.smart_filters:
+                for chart in self.chart.charts:
+                    chart.source.on_change('selected', self, 'on_selection_changed')
+                self.chart.session.poll_document(self.chart.doc)
+
         except TypeError:
+            raise
             if not self.series:
                 series_list = ', '.join(self.chart.values.keys())
                 print(hm.ERR_MSG_TEMPL % series_list)
                 raise
 
         if self.sync_with_source:
-            print("animating... press ctrl-C to stop")
             keep_source_input_sync(self.input, self.update_source, self.last_byte)
 
+    def on_copy(self, *args, **kws):
+        print("COPYING CONTENT!")
+        # TODO: EXPERIMENTAL!!! THIS EXPOSE MANY SECURITY ISSUES AND SHOULD
+        #       BE REMOVED ASAP!
+        txt = ''
+        for rowind in self.current_selection:
+            row = self.source.iloc[rowind]
+            txt += u"%s\n" % (u",".join(str(row[c]) for c in self.columns))
+
+        os.system("echo '%s' | pbcopy" % txt)
```

Comment:
Yeap... it was a last minute addition :-)

>> As an aside I don't think the clipboard copy is really even needed, the best thing would be for the selection to pass through stdout so it could be piped to another process

Mmm.. maybe. I thought that clipboard was useful for users wanting to do some filtering/analysis and then paste to excel, text editor or wherever they want.

Shall we to change to `sys.stdout.write(line)` in this PR? Maybe later?